src/cachehash/main.py

```diff
@@ -122,18 +128,55 @@ def set(self, file_path: Union[str, Path], values: dict):
         if not file_path.exists():
             raise ValueError(f"{file_path} does not exist")
 
-        if isinstance(values, dict):
-            values = json.dumps(values, indent=4)
-        else:
-            raise ValueError("Must pass values as a dict")
+        values = self._values_to_str(values)
 
         hash = self.get_hash(file_path)
-        self.query(
-            "insert_record",
+        existing_record = self.query(
+            "get_record",
             {
                 "key": fp,
-                "hash": str(hash),
-                "value": values,
-            }
-        )
+            },
+        ).fetchone()
+        if existing_record is not None:
+            self.query(
+                "update_record",
+                {
+                    "key": fp,
+                    "hash": str(hash),
+                    "value": values,
+                },
+            )
+        else:
+            self.query(
+                "insert_record",
+                {
+                    "key": fp,
+                    "hash": str(hash),
+                    "value": values,
+                },
+            )
         self.db.commit()
+
+    def _values_to_str(self, values):
+        if isinstance(values, (dict, list)):
+            return json.dumps(values)
+        return str(values)
+
+    def _values_from_str(self, values):
```

Comment:
I don't think we need all this
![image](https://github.com/user-attachments/assets/f4c673a3-e3c9-4070-9ccd-d14d58ead708)

Reply:
I was getting an error on the string unit test that JSON.loads didn't like it. But I'll test it again to see if I can simplify this

Reply:
I tried this:
```
    def _values_from_str(self, values):
        try:
            return json.loads(json.dumps(values))
        except json.JSONDecodeError:
            pass
        return values
```
And that failed to return a Dict that had been set.

So I tried this too:
```
    def _values_from_str(self, values):
        try:
            return json.loads(values)
        except json.JSONDecodeError:
            pass
        return values
```
And that returned an error for Complex values. It returned the complex value as a string:
```
E       AssertionError: Invalid Complex value
E       assert '(42+42j)' == (42 + 42j)
```